### PR TITLE
PUBDEV-3568: h2o.max returns NaN even when na.rf condition is set to TRUE

### PR DESCRIPTION
**Pull request by navdeep-G**
* Update h2o.* primitives to account for na.rm. 
* Also, remove if() clauses in h2o.* primitives as they are not needed since these methods are base primitives for h2o. These wrapper ,i.e. h2o.*, are merely here for completeness and consistency.

---
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/370)

### CODE REVIEW

File: h2o-r/h2o-package/R/frame.R [hunk 1]
```diff
@@ -2636,8 +2606,7 @@ h2o.all <- function(x) {
 #' @seealso \code{\link[base]{all}} for the base R implementation.
 #' @export
 h2o.any <- function(x) {
-  if( is.H2OFrame(x) ) all(x)
-  else all(x)
+  all(x)
```
Comment (jangorecki) [inline h2o-r/h2o-package/R/frame.R]: Any idea why it was `all(x)` before your commit? looks like it should be `any`

File: h2o-r/h2o-package/R/frame.R [hunk 2]
```diff
@@ -2439,8 +2424,7 @@ h2o.print <- function(x,n=6L) {
 #' @param cols Print the per-column str for the H2OFrame
 #' @export
 h2o.str <- function(object, ..., cols=FALSE) {
-  if(is.H2OFrame(object)) str(object, ..., cols)
-  else str(object, ..., cols)
+  str(object, ..., cols)
```
Comment (jangorecki) [inline h2o-r/h2o-package/R/frame.R]: AFAIK argument `cols` will be ignored, arguments passed after `...` should always be named.
```r
g = function(x, ..., cols=TRUE) {
    print(cols)
    return(invisible(x))
}
f = function(x, ..., cols=FALSE) {
    g(x, ..., cols)
}
f(1)
#[1] TRUE
f(1, cols=FALSE)
#[1] TRUE
```
it was a bug before :)

File: h2o-r/h2o-package/R/frame.R [hunk 3]
```diff
@@ -2428,8 +2414,7 @@ h2o.ascharacter <- function(x) {
 #' @param ... Further arguments to be passed from or to other methods.
 #' @export
 h2o.print <- function(x,n=6L) {
-  if(is.H2OFrame(x)) print(x,n)
-  else print(x,n)
+  print(x,n)
```
Comment (jangorecki) [inline h2o-r/h2o-package/R/frame.R]: `n` is passed to `...` generic print, then matched to `n` in h2o print method, no issue here, but name `n` explicitly: `print(x, n = n)` so argument passed to dots won't rely on position.

**Comment by h2o-ops** [comment]
Tests passed!

File: h2o-r/h2o-package/R/frame.R (hunk 3 above)
Comment (navdeep-G) [inline h2o-r/h2o-package/R/frame.R]: Good suggestion.

File: h2o-r/h2o-package/R/frame.R (hunk 1 above)
Comment (navdeep-G) [inline h2o-r/h2o-package/R/frame.R]: I can change that. Might've been a typo.

File: h2o-r/h2o-package/R/frame.R [hunk 4]
```diff
@@ -2700,8 +2665,7 @@ h2o.ncol <- function(x) {
 #' @seealso \code{\link[base]{range}} for the base R implementation.
 #' @export
 h2o.range <- function(x,na.rm = FALSE,finite = FALSE) {
-  if( is.H2OFrame(x) ) range(x,na.rm,finite)
-  else range(x,na.rm,finite)
+  range(x,na.rm = na.rm,finite)
```
Comment (jangorecki) [inline h2o-r/h2o-package/R/frame.R]: `finite` argument is probably not used, see [range.H2OFrame](https://github.com/h2oai/h2o-3/blob/97e4d2998ae8cfb7f769bf946f9ac51c09eb0f0f/h2o-r/h2o-package/R/frame.R#L3301). Will create separate issue, as this is not directly related to this PR.

**Comment by navdeep-G** [comment]
@h2o-ops please test

**Comment by h2o-ops** [comment]
Tests passed!
